nest/test/AggregateTokenCredbull.t.sol

```diff
@@ -0,0 +1,206 @@
+    function unlockRequestAmount(address owner, uint256 requestId) external view returns (uint256);
+    function claimableRedeemRequest(uint256 requestId, address controller) external view returns (uint256);
+}
+
+contract AggregateTokenCredbullTest is Test {
+    // Contract addresses - testnet addresses
+    // run with: forge test -vvvv --rpc-url $PLUME_RPC_URL --match-contract AggregateTokenCredbullTest
+
+    address constant USDT_ADDRESS = 0x2413b8C79Ce60045882559f63d308aE3DFE0903d;
+    address constant USDC_ADDRESS = 0x401eCb1D350407f13ba348573E5630B83638E30D;
+    address constant CREDBULL_ADDRESS = 0x4B1fC984F324D2A0fDD5cD83925124b61175f5C6;
+
+    ICredbullVault public constant CREDBULL_VAULT = ICredbullVault(CREDBULL_ADDRESS);
+    IUSDT public constant USDT = IUSDT(USDT_ADDRESS);
+    IUSDC public constant USDC = IUSDC(USDC_ADDRESS);
+
+    // Test parameters
+    uint256 constant TEST_AMOUNT = 1e6; // $1
+    uint256 constant BASE = 1e18;
+
+    // Contract instances
+    AggregateToken public token;
+    address public account;
+
+    function setUp() public {
+        // needs to have some USDC
+        uint256 privateKey = vm.envUint("PRIVATE_KEY");
+        account = vm.addr(privateKey);
+        address controller = vm.addr(privateKey);
+
+        console.log("Setting up test with account:", account);
+        console.log("Test amount:", TEST_AMOUNT);
+
+        vm.startBroadcast(privateKey);
+
+        // First check if we need to become admin
+        bytes32 adminRole = CREDBULL_VAULT.DEFAULT_ADMIN_ROLE();
+        bool isAdmin = CREDBULL_VAULT.hasRole(adminRole, controller);
+
+        if (!isAdmin) {
+            // For testing, we can impersonate the current admin to grant our account admin rights
+            // First, get the current admin
+            address currentAdmin = CREDBULL_VAULT.getRoleMember(adminRole, 0);
+            vm.stopBroadcast();
+            vm.startPrank(currentAdmin);
+            CREDBULL_VAULT.grantRole(adminRole, controller);
+            vm.stopPrank();
+            vm.startBroadcast(privateKey);
+        }
+
+        // Deploy implementation
+        AggregateToken implementation = new AggregateToken();
+        console.log("Implementation deployed at:", address(implementation));
+
+        // Initialize with USDC as asset token
+        bytes memory initData = abi.encodeCall(
+            AggregateToken.initialize,
+            (
+                account, // owner
+                "Test Aggregate USD", // name
+                "tAUSD", // symbol
+                IComponentToken(USDC_ADDRESS), // USDC as asset token
+                BASE, // ask price
+                BASE // bid price
+            )
+        );
+
+        // Deploy proxy
+        AggregateTokenProxy proxy = new AggregateTokenProxy(address(implementation), initData);
+        console.log("Proxy deployed at:", address(proxy));
+
+        token = AggregateToken(address(proxy));
```

Comment:
Can you use the existing NEV AggregateToken that's already deployed at [0x659619AEdf381c3739B0375082C2d61eC1fD8835](https://test-explorer.plumenetwork.xyz/address/0x659619AEdf381c3739B0375082C2d61eC1fD8835)?

Reply:
No need to do this - we can deploy a new one when testing